openapi.yaml

```diff
@@ -268,19 +713,95 @@ components:
             type: string
             maxLength: 128
           nullable: true
-    LearningResource:
+    LearningPath:
+      type: object
+      description: Serializer for the LearningPath model
+      properties:
+        id:
+          type: integer
+          readOnly: true
+        item_count:
+          type: integer
+          description: Return the number of items in the list
+          readOnly: true
+        author:
+          type: integer
+      required:
+      - author
+      - id
+      - item_count
+    LearningPathRelationship:
+      type: object
+      description: Specialized serializer for a LearningPath relationship
+      properties:
+        id:
+          type: integer
+          readOnly: true
+        resource:
+          allOf:
+          - $ref: '#/components/schemas/LearningResource'
+          readOnly: true
+          nullable: true
+        relation_type:
+          type: string
+          default: LEARNING_PATH_ITEMS
+        position:
+          type: integer
+          maximum: 2147483647
+          minimum: 0
+        parent:
+          type: integer
+        child:
+          type: integer
+      required:
+      - child
+      - id
+      - parent
+      - resource
+    LearningPathResource:
       type: object
-      description: Full serializer for LearningResource
+      description: CRUD serializer for LearningPath resources
       properties:
         id:
           type: integer
           readOnly: true
+        title:
+          type: string
+          maxLength: 256
+        description:
+          type: string
+          nullable: true
+        readable_id:
+          type: string
+          maxLength: 128
         topics:
           type: array
           items:
             $ref: '#/components/schemas/LearningResourceTopic'
+          nullable: true
```

Comment:
Currently the OpenAPI spec and generated typescript types say that full `LearningResourceTopic` objects are:
- returned in GET requests
- expected for POST/PATCH requests

The POST/PATCH requests actually accept either fully LearningResourceTopic objects OR just their ids. See

Reply:
Opened #54 (perhaps prematurely) about this.